lib/alipay/sign.rb

```diff
@@ -47,6 +56,26 @@ def self.verify?(params, options = {})
       end
     end
 
+    def self.app_pay_verify?(params, options = {})
+      params = Utils.stringify_keys(params)
+
+      sign_type = params.delete('sign_type')
+      sign = params.delete('sign')
+      string = params_to_string(params)
+
+      case sign_type
+      when 'MD5'
+        key = options[:key] || Alipay.key
+        MD5.verify?(key, string, sign)
+      when 'RSA'
+        RSA.verify?(ALIPAY_RSA_APP_PAY_PUBLIC_KEY, string, sign)
```

Comment:
Oh, I miss the filename, this sign logic should move to `alipay/app/sign.rb`, like `alipay/wap/sign.rb`.

And service methods need a unit test.